### PR TITLE
8302781: CDS archive heap not reproducible after JDK-8296344

### PR DESCRIPTION
**Pull request by iklam**
[JDK-8296344](https://bugs.openjdk.org/browse/JDK-8296344) changed the order of when `oopDesc::identity_hash()` is called for the archived heap object.

This one-liner fix restores the order, so that we initialize the `identity_hash` part of the archived object headers predictably.

Please see [my comment in the JBS report](https://bugs.openjdk.org/browse/JDK-8302781?focusedCommentId=14561163&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14561163) for details.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302781](https://bugs.openjdk.org/browse/JDK-8302781): CDS archive heap not reproducible after JDK-8296344

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to [0eb32cf9](https://git.openjdk.org/jdk/pull/12628/files/0eb32cf9670f4f7f53c83c2a7ac9fb6f9b445e79)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12628/head:pull/12628` \
`$ git checkout pull/12628`

Update a local copy of the PR: \
`$ git checkout pull/12628` \
`$ git pull https://git.openjdk.org/jdk pull/12628/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12628`

View PR using the GUI difftool: \
`$ git pr show -t 12628`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12628.diff">https://git.openjdk.org/jdk/pull/12628.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/cds/heapShared.cpp [hunk 1]
```diff
@@ -286,6 +286,11 @@ bool HeapShared::archive_object(oop obj) {
     count_allocation(obj->size());
     ArchiveHeapWriter::add_source_obj(obj);
 
+    // The archived objects are discovered in a predictable order. Compute
+    // their identity_hash() as soon as we see them. This ensure that the
```
Comment (dcubed-ojdk) [inline src/hotspot/share/cds/heapShared.cpp:290]: nit typo: s/This ensure/This ensures/

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. One very minor typo. This is a trivial fix and does not
need to wait 24 hours.

**Comment by iklam** [comment]
Thanks @dcubed-ojdk for the review.
Passed builds-tier5 (where the original failure was detected). Also passed tiers 1 and 2.
/integrate

File: src/hotspot/share/cds/heapShared.cpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/cds/heapShared.cpp]: Fixed.

**Pull request closed without merging**